src/compiler/checker.ts

```diff
@@ -3981,8 +3981,15 @@ namespace ts {
                 return res;
             }
             const candidates = mapDefined(symbol.declarations, d => {
-                if (!isAmbientModule(d) && d.parent && hasNonGlobalAugmentationExternalModuleSymbol(d.parent)) {
-                    return getSymbolOfNode(d.parent);
+                if (!isAmbientModule(d) && d.parent){
+                    // direct children of a module
+                    if (hasNonGlobalAugmentationExternalModuleSymbol(d.parent)) {
+                        return getSymbolOfNode(d.parent);
+                    }
+                    // export ='d member of an ambient module
+                    if (isModuleBlock(d.parent) && d.parent.parent && resolveSymbol(getSymbolOfNode(d.parent.parent).exports?.get(InternalSymbolName.ExportEquals)) === symbol) {

```

Comment:
This is almost an inlining of `resolveExternalModuleSymbol`, but you omit `getMergedSymbol` (and some logic that I believe is JS-specific merging that wouldn’t run here). Do you need to get the merged symbol of any of these?

Reply:
While you're right that some of that cjs stuff will never apply here, you're also right that it's probably better to just use `resolveExternalModuleSymbol`, which it now does.